core/src/main/scala/org/apache/spark/ui/SparkUI.scala

```diff
@@ -86,6 +86,7 @@ private[spark] class SparkUI private (
         .orElse(store.environmentInfo().systemProperties.toMap.get("user.name"))
         .getOrElse("<unknown>")
     } catch {
+      case sparkException: SparkException => "<unknown>"
       case _: NoSuchElementException => "<unknown>"
```

Comment:
this is needed since store.environmentInfo() , can throw NoSuchElementException

Reply:
cc: @sarutak

Reply:
nit `case _: SparkException => "<unknown>"`

Reply:
done